### DISCUSSION TITLE
View - 4. Feature List

### DISCUSSION THREAD

**Discussion by jin-Pro** [githru/githru-vscode-ext#16] [View]
## 1. Temporal Filter

<img src="https://user-images.githubusercontent.com/70205497/185013647-ae80c4a0-b957-4541-b0fe-99c2e8aa6a4a.jpeg" 
 width="300" height="300" />

시간에 따라 
- commit 수 
- cloc(count line of code 코드 라인 변경 수)
에 대한 icicle 차트 2개를 나타냅니다.

구간을 선택해주면 Vertical Cluster List와 Statistics 데이터에 반영됩니다.

## 2. Vertical Cluster List

<img src="https://user-images.githubusercontent.com/70205497/185014817-ec1610d0-8095-49ff-802c-b137c62ac475.jpeg" 
 width="300" height="300" />

왼쪽의 Graph (CSM) 와 우측의 details가 존재합니다.
지정된 height을 기준으로 scroll 기능이 존재합니다.

graph는 커밋과 클러스터 두가지 버전으로 존재하며
detail에는 커밋 작성자와 제목이 들어갑니다.

detail을 클릭하면 statistics 내용에 반영됩니다.
detail을 클릭하면 아래 Detail에 반영됩니다.

## 3. Statistics

<img src="https://user-images.githubusercontent.com/70205497/185014845-43934cae-6e8a-46d9-a44b-e000a4657566.jpeg" 
 width="300" height="300" />

Temporal Filter에 지정된 구간에 존재하는 데이터를 기준으로 통계자료를 나타냅니다.
또한,
Vertical Cluster List의 detail을 클릭하면 해당 commit의 통계자료를 나타냅니다.

## 4. Detail

<img src="https://user-images.githubusercontent.com/70205497/185014909-496ab04a-2cfd-4bb6-9570-6f630f89e170.jpeg" 
 width="300" height="300" />

pr정보, commit 정보 등 내용이 들어갑니다.

**Comment by jin-Pro** [comment]
1. Temporal Filter 

2개의 icicle ( commit , cloc ) 이 렌더 되는게 맞을까요??

@ytaek

**Reply by ytaek** [reply]
넵 일단은 githru.org 처럼 그렇게 하면 좋겠네요.

**Reply by jeonghye-choi** [reply]
@ytaek '시간에 따라' 라는 말이 시간에 따라 그래프가 동적으로 변해야한다는 의미일까요?
git graph 같은 경우에는 커밋을 한다던가, 머지 또는 push 와 같은 변경사항이 생겼을 때 그래프에 바로 반영이 되는 기능이 있는데, 
그런 동적인 반응을 의미하는 걸까요??

**Reply by jeonghye-choi** [reply]
회의 때([220816회의](https://github.com/githru/githru-vscode-ext/wiki/View-%7C-220816-Meeting-Note)), Temperal Filter가 line chart로 구현되는 걸로 이해했는데, icicle 구현인걸까요??

**Reply by ytaek** [reply]
> @ytaek '시간에 따라' 라는 말이 시간에 따라 그래프가 동적으로 변해야한다는 의미일까요? git graph 같은 경우에는 커밋을 한다던가, 머지 또는 push 와 같은 변경사항이 생겼을 때 그래프에 바로 반영이 되는 기능이 있는데, 그런 동적인 반응을 의미하는 걸까요??

Temporal Filter는 우선 전체 commit 기간에 대하여 보여주는 것이라, 
전체 데이터가 refresh 되지 않는 이상 바뀔 일이 없다고 보시면 됩니다.
(global temporal filter라고 따로 있긴 한데, 아직 없다고 보시면 되요)


> 회의 때([220816회의](https://github.com/githru/githru-vscode-ext/wiki/View-%7C-220816-Meeting-Note)), Temperal Filter가 line chart로 구현되는 걸로 이해했는데, icicle 구현인걸까요??

기존과 동일한 line chart 맞습니다~.

**Comment by jin-Pro** [comment]
2. Vertical Cluster List

detail을 클릭하였을 때
하단의 4. Detail 변경 이외에도

detail이 토글 목록처럼 펼쳐진다고 생각하면 될까요? ( cluster 의 경우만? )

**Reply by jejecrunch** [reply]
저는 해당 부분을 그렇게 이해했습니다 !

**Reply by ytaek** [reply]
네, 제 의도는 그렇습니다 : -) 
펼쳐지는 걸로.

펼쳐지게 하는 interaction은 graph node에 넣어야 할지, 전체로 넣어야 할 지 조금 애매합니다.
selection 하는 interaction과 구분해야 할 수도 있어서?
혹은 shift + click으로 다중 선택이 되게 할 수도 있겠네요.

**Reply by jin-Pro** [reply]
이부분도 18일 ( 금일 ) 회의할때 얘기해보면 좋을 것 같네요!

**Comment by jin-Pro** [comment]
3. Statistics

여기에 들어갈 통계 자료는 무엇을 나타내줘야할지 정해야하는게 맞을까요??

**Reply by jejecrunch** [reply]
특정 기간에 대한 통계기 때문에 무슨 요일에 PR이 많았는지, COL이 많았는지 이런 걸 보여주는 파트라고 생각했습니다.

**Reply by ytaek** [reply]
githru.org 베이스로

1. author별 CLOC, Commit #.
2. File Icicle Tree (아마도 cell의 높이에 cloc나 commit# 가 encoding되겠죠)
를 기본으로,

PR (closed/merged/open 등), 
commit type (아마도 keyword 베이스로 분류할 것 같네요),

등등이 나오면 좋겠습니다.

**Comment by jin-Pro** [comment]
4. Detail

여기에 들어갈 graph detail 데이터는 pr정보 commit 정보 이외에도 어떠한 데이터를 나타내줄지 정하면 되는걸까요??

**Reply by jejecrunch** [reply]
우선은 PR정보 Commit 정보만 띄워주는 것을 0.1에 하는 게 더 시간적으로 생각했을 때 할 수 있을 것 같습니다 !

**Reply by ytaek** [reply]
PR의 description + Commit List만 보여줘도 지금은 괜찮을 것 같네요.
(그리고 PR이 여러개 포함될 수도 있어서, 그걸 어떻게 aggregation 해줘야 하는지도 고민해볼 필요는 있겠습니다)
